ironfish-zkp/src/circuits/mint_asset.rs

```diff
@@ -22,6 +26,39 @@ pub struct MintAsset {
     pub public_key_randomness: Option<jubjub::Fr>,
 }
 
+impl MintAsset {
+    pub fn write<W: Write>(&self, mut writer: W) -> std::io::Result<()> {
+        if let Some(proof_generation_key) = &self.proof_generation_key {
```

Comment:
Nice, thanks for the tip. I wouldn't have guessed it would be copying here, thought it would just be unwrapping the optional. No harm in updating to this syntax